src/lib/protocols/cassandra.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * cassandra.c
+ *
+ * Copyright (C) 2021 by Lucas Santos <lfneiva.santos@gmail.com>
+ *
+ * This file is part of nDPI, an open source deep packet inspection
+ * library based on the OpenDPI and PACE technology by ipoque GmbH
+ *
+ * nDPI is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * nDPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with nDPI.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include "ndpi_protocol_ids.h"
+#include "ndpi_api.h"
+
+#define NDPI_CURRENT_PROTO NDPI_PROTOCOL_CASSANDRA
+#define CASSANDRA_HEADER_LEN 9
+#define CASSANDRA_MAX_BODY_SIZE 268435456 //256MB (256 * 1024^2)
+
+enum cassandra_version
+{
+  CASSANDRA_V1_REQUEST = 0x01,
+  CASSANDRA_V1_RESPONSE = 0x81,
+  CASSANDRA_V2_REQUEST = 0x02,
+  CASSANDRA_V2_RESPONSE = 0x82,
+  CASSANDRA_V3_REQUEST = 0x03,
+  CASSANDRA_V3_RESPONSE = 0x83,
+  CASSANDRA_V4_REQUEST = 0x04,
+  CASSANDRA_V4_RESPONSE = 0x84
+};
+
+enum cassandra_opcode
+{
+  ERROR = 0x00,
```

Comment:
It is probably better to prefix enum constants with something less generic e.g. `CASSANDRA_`, because something like `ERROR` might be already defined by MingW.

```suggestion
  CASSANDRA_ERROR = 0x00,
```

and so on for the rest of the enum constants.

Reply:
I agree, it was an oversight of my part. Just pushed the changes!